From 89c5a64a676f7193e0c52cfd802e9ca5129577b4 Mon Sep 17 00:00:00 2001
From: Zhao Qiang <qiang.zhao@nxp.com>
Date: Wed, 13 Jan 2016 11:16:50 +0800
Subject: [PATCH 0422/1383] QE: Add "ifdef CONFIG_QUICC_ENGINE" for qe

qe and qe_ic shouldn't be initialized when QUICC_ENGINE
is not selected.

Signed-off-by: Zhao Qiang <qiang.zhao@nxp.com>
[Original patch taken from QorIQ-SDK-V2.0-20160527-yocto]
Signed-off-by: Yanjiang Jin <yanjiang.jin@windriver.com>
---
 arch/powerpc/platforms/85xx/corenet_generic.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/arch/powerpc/platforms/85xx/corenet_generic.c b/arch/powerpc/platforms/85xx/corenet_generic.c
index 123b9e2..7e051f9 100644
--- a/arch/powerpc/platforms/85xx/corenet_generic.c
+++ b/arch/powerpc/platforms/85xx/corenet_generic.c
@@ -41,7 +41,9 @@ void __init corenet_gen_pic_init(void)
 	unsigned int flags = MPIC_BIG_ENDIAN | MPIC_SINGLE_DEST_CPU |
 		MPIC_NO_RESET;
 
+#ifdef CONFIG_QUICC_ENGINE
 	struct device_node *np;
+#endif
 
 	if (ppc_md.get_irq == mpic_get_coreint_irq)
 		flags |= MPIC_ENABLE_COREINT;
@@ -51,12 +53,14 @@ void __init corenet_gen_pic_init(void)
 
 	mpic_init(mpic);
 
+#ifdef CONFIG_QUICC_ENGINE
 	np = of_find_compatible_node(NULL, NULL, "fsl,qe-ic");
 	if (np) {
 		qe_ic_init(np, 0, qe_ic_cascade_low_mpic,
 				qe_ic_cascade_high_mpic);
 		of_node_put(np);
 	}
+#endif
 }
 
 /*
@@ -114,9 +118,11 @@ static const struct of_device_id of_device_ids[] = {
 	{
 		.compatible	= "fsl,qoriq-pcie-v3.0",
 	},
+#ifdef CONFIG_QUICC_ENGINE
 	{
 		.compatible	= "fsl,qe",
 	},
+#endif
 	{
 		.compatible    = "fsl,fman",
 	},
-- 
2.8.1

